corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -353,9 +357,23 @@ def build_query_prompts(self):
                     )
                     for i, validation in enumerate(prop.validations)
                 ]
+            if prop.group_key:
+                kwargs['group_key'] = prop.group_key
             prompts.append(QueryPrompt(**kwargs))
         return prompts
 
+    def build_query_prompt_groups(self):
+        prompts = []
+        prompt_group_properties = [prop for prop in self.module.search_config.properties if prop.is_group]
```

Comment:
Is there a way to filter out group properties vs prompt properties before calling their respective functions? Not even sure if it makes sense to do so, just thinking about iterating over the same set of items multiple times again.

Reply:
It's possible. They're called only in [build_remote_request_queries](https://github.com/dimagi/commcare-hq/blob/e47af0700c84e0ba5effc23321d3ec57f40cb7d3/corehq/apps/app_manager/suite_xml/post_process/remote_requests.py#L204) so the filtering could be done there first and a parameter for the respective function. 

But I would prefer looping over twice than adding a parameter for each function.